mmtrack/datasets/vot_dataset.py

```diff
@@ -75,40 +121,53 @@ def evaluate(self, results, metric=['track'], logger=None, interval=None):
             if metric not in allowed_metrics:
                 raise KeyError(f'metric {metric} is not supported.')
 
+        # get all test annotations
+        # annotations are in list[list[ndarray]] format
+        annotations = []
+        for video_ind in range(len(self.data_infos)):
+            bboxes = self.get_ann_infos_from_video(video_ind)['bboxes']
+            annotations.append(bboxes)
+
+        # tracking_bboxes converting code
         eval_results = dict()
         if 'track' in metrics:
-            assert len(self.data_infos) == len(results['track_bboxes'])
+            assert len(self) == len(
+                results['track_bboxes']
+            ), f"{len(self)} == {len(results['track_bboxes'])}"
             print_log('Evaluate VOT Benchmark...', logger=logger)
-            inds = []
+            track_bboxes = []
+            start_ind = end_ind = 0
             videos_wh = []
-            ann_infos = []
-            for i, info in enumerate(self.data_infos):
-                if info['frame_id'] == 0:
-                    inds.append(i)
-                    videos_wh.append((info['width'], info['height']))
+            for data_info in self.data_infos:
+                num = data_info['end_frame_id'] - data_info[
+                    'start_frame_id'] + 1
+                end_ind += num
 
-                ann_infos.append(self.get_ann_info(info))
-
-            num_vids = len(inds)
-            inds.append(len(self.data_infos))
-            track_bboxes = []
-            annotations = []
-            for i in range(num_vids):
                 bboxes_per_video = []
-                for bbox in results['track_bboxes'][inds[i]:inds[i + 1]]:
+                # results are in dict(track_bboxes=list[ndarray]) format
```

Comment:
`track_bboxes=list[list[ndarray]]`

Reply:
`results` is the variable in the next line.
This `results` is the original track results, not the inputs of `eval_sot_eao`

Reply:
move the comments in the above